stubs/nvml_stubs.c

```diff
@@ -674,7 +674,7 @@ stub_nvml_get_pgpu_vgpu_compatibility(value ml_interface,
 
     size_t compatSize = sizeof(nvmlVgpuPgpuCompatibility_t);
     ml_vgpu_pgpu_compat_meta = caml_alloc_string(compatSize);
-    memcpy(String_val(ml_vgpu_pgpu_compat_meta),
+    memcpy((char *) String_val(ml_vgpu_pgpu_compat_meta),
```

Comment:
same on all of these, use Bytes_val (although in theory that might still be incompatible because we alloc as string, then change it as bytes and return as string again without using Bytes.unsafe_to_string.

Reply:
Even better would be to use https://v2.ocaml.org/manual/intfc.html#ss:c-block-allocation `caml_alloc_initialized_string` and give it the size and const char* where to copy it from (we can't use `caml_copy_string` because I assume this is actually binary and might contain embedded `\0`)

Reply:
The `const` change is just a drive-by shooting. The main theme is the protocol how the API is used.

Reply:
Yes, the rest of the change here looks fine.

Reply:
I've removed the `memcpy` - but this still requires casts. I do think it is the cleaner solution, though. I'll squash commits once this is reviewed.